infra/build/functions/sync/main.py

```diff
@@ -35,41 +41,155 @@ class GitAuth(ndb.Model):
   access_token = ndb.StringProperty()
 
 
-def sync_projects(projects):
+def create_scheduler(cloud_scheduler_client, project_name, schedule):
+  """Creates schedulers for new projects."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project_name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': bytes(project_name, 'utf-8')
+      },
+      'schedule': schedule
+  }
+
+  try:
+    cloud_scheduler_client.create_job(parent, job)
+    return True
+
+  except exceptions.GoogleAPICallError as error:
+    print(error.message)
+
+  except exceptions.RetryError as error:
```

Comment:
Looking at https://googleapis.dev/python/google-api-core/latest/exceptions.html, RetryError is just a subclass of GoogleAPICallError. There's no reason to distinguish them here. 

That said, this doesn't seem like the right approach to handling errors. We shouldn't handle them here, and instead bubble them back up to the caller.

With the current implementation, you update and delete Project entities even if these operations fail (e.g. due to some transient error), which mean that they will never get retried again and we end up in an inconsistent state.

Instead, you should restructure your operations such that the main loop looks something like this:

```python
for project in projects_to_remove:  # type changed here from your current code to actual entity type.
  try:
    delete_scheduler(...)
    project.key.delete()
  except exceptions.GoogleAPICallError as e:
    logging.error('Scheduler deletion for %s failed with %s', project.name, e)
```

Similarly for update case, don't `put()` unless the operation suceeds.